eng/common/pipelines/templates/jobs/archetype-spec-gen-sdk.yml

```diff
@@ -148,6 +151,14 @@ jobs:
           sdk_gen_info="$sdk_gen_info '${{ parameters.ConfigPath }}',"
         fi
 
+        if [ "$(Build.Reason)" = "PullRequest" ]; then
```

Comment:
Any reason not to do this above? BTW if you set the variable in the same step you cannot use it in that same script block it can only be used in a future step.

Reply:
It needs to be after the sparse checkout step. So either the step of `Set SpecRepoCommit variable` or this step. I chose this step since it will only be used in the `code push` step.

Reply:
I don't think you set and consume the variable in the same step so you will at least need to split.